tv_apps/browser/test/unit/sync/settings_test.js

```diff
@@ -0,0 +1,278 @@
+      expect(onsyncchange).to.be.a('function');
+      expect(visibilityListener).to.be.a('function');
+    });
+
+    test('should observe settings', function() {
+      expect(observeStub.calledTwice).to.be.equal(true);
+      expect(observers[HISTORY_SETTING]).to.be.a('function');
+      expect(observers[BOOKMARKS_SETTING]).to.be.a('function');
+    });
+
+    test('DOM initial state', function() {
+      expect(subject.area).to.be.an('object');
+      expect(subject.elements.enabled).to.be.an('object');
+      expect(subject.elements.disabled).to.be.an('object');
+      expect(subject.listeners).to.be.an('object');
+      expect(subject.listeners.size).to.be.equals(0);
+      expect(subject.collections).to.be.an('object');
+      expect(subject.collections.size).to.be.equals(0);
+      expect(subject.screen).to.be.equals(null);
+      expect(subject.state).to.be.equals(null);
+    });
+  });
+
+  suite('visibilitychange', function() {
+    setup(function() {
+      getInfoSpy.reset();
+    });
+
+    teardown(function() {
+      getInfoSpy.reset();
+    });
+
+    test('should refresh on visibity change', function() {
+      expect(getInfoSpy.called).to.be.equal(false);
+      visibilityListener();
+      expect(getInfoSpy.called).to.be.equal(true);
+    });
+  });
+
+  suite('Disabled', function() {
+    suiteSetup(function() {
+      onsyncchange({
+        state: 'disabled'
+      });
+    });
+
+    suiteTeardown(function() {
+      showScreenSpy.reset();
+    });
+
+    test('should show disabled screen', function() {
+      expect(showScreenSpy.calledOnce).to.be.equal(true);
+      expect(showScreenSpy.args[0][0]).to.equal('disabled');
+      expect(subject.elements.enabled.hidden).to.be.equal(true);
+      expect(subject.elements.disabled.hidden).to.be.equal(false);
+    });
+
+    test('should have signin button', function() {
+      expect(subject.elements.signInButton).to.be.an('object');
+      expect(subject.elements.signInButton.getAttribute('data-l10n-id'))
+        .to.equals('fxsync-create-account-or-sign-in');
+    });
+  });
+
+  suite('Enabled', function() {
+    suiteSetup(function() {
+      onsyncchange({
+        state: 'enabled',
+        user: 'pepito'
+      });
+    });
+
+    suiteTeardown(function() {
+      showScreenSpy.reset();
+    });
+
+    test('should show enabled screen', function() {
+      expect(showScreenSpy.calledOnce).to.be.equal(true);
+      expect(showScreenSpy.args[0][0]).to.equal('enabled');
+      expect(subject.elements.enabled.hidden).to.be.equal(false);
+      expect(subject.elements.disabled.hidden).to.be.equal(true);
+    });
+
+    test('should show user', function() {
+      expect(subject.elements.signedInAs).to.be.an('object');
+      expect(subject.elements.signedInAs.getAttribute('data-l10n-id'))
+        .to.equals('fxsync-signed-in-as');
+      expect(subject.elements.signedInAs.getAttribute('data-l10n-args'))
+        .to.equals('{"email":"pepito"}');
+
```

Comment:
This would need to be changed once the ```Sign in as``` text gets removed according to the spec. We can do it in a follow bug if you'd like.